system/src/system_sleep.cpp

```diff
@@ -89,6 +89,8 @@ int system_sleep_ext(const hal_sleep_config_t* config, hal_wakeup_source_base_t*
         if (system_sleep_network_suspend(NETWORK_INTERFACE_CELLULAR)) {
             cellularResume = true;
         }
+        // There might be up to 30s delay to turn off the modem for particular platforms.
+        network_wait_modem_off(NETWORK_INTERFACE_CELLULAR, 60000/*ms*/, nullptr);
```

Comment:
This will affect all sleep timings when using STOP mode with default settings.

1. Let's discuss more if we can potentially do this in a manner which does not "break" things
2. We should look into potentially improving power off timings e.g. by using `AT+CPWROFF` or others

Reply:
I didn't understand that why it breaks the sleep timing, especially for the STOP mode. I think if the device is requested to enter STOP mode without specifying Cellular interface as wake-up source, we should also wait util the modem is completely turned off.

Reply:
Which might be up to tens of seconds on certain platforms. So, the actual sleep will now suddenly be postponed by that amount of time after we introduce this change. I would consider this a change in behavior.